sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java

```diff
@@ -87,6 +90,98 @@ void setup() {
         this.soapUtil = riInjector.getInstance(SoapUtil.class);
     }
 
+    @Test
+    @TestIdentifier(EnabledTestConfig.BICEPS_C5)
+    @TestDescription("Starting from the initially retrieved mdib, applies each episodic report to the mdib and checks"
+            + " for each AbstractDescriptor contained in a DescriptionModificationReport "
+            + "that it was inserted or deleted or udpdated by changing"
+            + "at least one child or attribute.")
+    @RequirePrecondition(simplePreconditions = ConditionalPreconditions.DescriptionChangedPrecondition.class)
+    void testRequirementC5() throws NoTestData, IOException {
+        final var mdibHistorian = mdibHistorianFactory.createMdibHistorian(
+                messageStorage, getInjector().getInstance(TestRunObserver.class));
+
+        final var acceptableSequenceSeen = new AtomicInteger(0);
+
+        try (final Stream<String> sequenceIds = mdibHistorian.getKnownSequenceIds()) {
+            sequenceIds.forEach(sequenceId -> {
+
+                // get relevant reports
+                try (final var reports = mdibHistorian.getAllReports(sequenceId)) {
+                    var first = mdibHistorian.createNewStorage(sequenceId);
+                    var second = mdibHistorian.createNewStorage(sequenceId);
+
+                    for (final Iterator<AbstractReport> iterator = reports.iterator(); iterator.hasNext(); ) {
+                        final AbstractReport report = iterator.next();
+
+                        if (report instanceof DescriptionModificationReport) {
+                            acceptableSequenceSeen.incrementAndGet();
+                            second = mdibHistorian.applyReportOnStorage(second, report);
+
+                            for (var reportPart : ((DescriptionModificationReport) report).getReportPart()) {
```

Comment:
You could remove the cast and create a variable in the instanceof check

Reply:
done